alfalfa_web/server/api.js

```diff
@@ -104,306 +92,280 @@ class AlfalfaAPI {
-    }
+    return this.getModels().map(this.formatModel);
+  };
+
+  getModels = async () => {
+    return this.models.find().toArray();
+  };
+
+  getModelById = async (modelId) => {
+    return this.models.findOne({ ref_id: modelId });
+  };
+
+  formatModel = (model) => {
+    const { ref_id: id, model_name: modelName, created, modified } = model;
+    return {
+      id,
+      modelName,
+      created,
+      modified
+    };
   };
 
-  createModel = async (modelID, modelName) => {
+  getModelDownloadPath = async (model) => {
+    const signedURL = await getSignedUrl(
+      this.s3,
+      new GetObjectCommand({
+        Bucket: process.env.S3_BUCKET,
+        Key: `uploads/${model.ref_id}/${model.model_name}`,
+        ResponseContentDisposition: `attachment; filename="${run.ref_id}.tar.gz"`
+      }),
+      {
+        expiresIn: 86400
+      }
+    );
+    return signedURL;
+  };
+
+  createModel = async (modelId, modelName) => {
     const datetime = new Date();
     await this.models.insertOne({
-      ref_id: modelID,
+      ref_id: modelId,
       model_name: modelName,
       created: datetime,
       modified: datetime,
       _cls: "Model"
     });
   };
 
-  createRunFromModel = async (modelID) => {
-    try {
-      const runID = uuidv1();
-      const model = await this.models.findOne({ ref_id: modelID });
-      if (model) {
-        const job = `alfalfa_worker.jobs.${model.model_name.endsWith(".fmu") ? "modelica" : "openstudio"}.CreateRun`;
-
-        const messageBody = {
-          job,
-          params: {
-            model_id: modelID,
-            run_id: runID
-          }
-        };
-
-        await this.sqs.send(
-          new SendMessageCommand({
-            MessageBody: JSON.stringify(messageBody),
-            QueueUrl: process.env.JOB_QUEUE_URL,
-            MessageGroupId: "Alfalfa"
-          })
-        );
-
-        return { runID };
+  createRunFromModel = async (model) => {
+    const runId = uuidv1();
+    const job = `alfalfa_worker.jobs.${model.model_name.endsWith(".fmu") ? "modelica" : "openstudio"}.CreateRun`;
+
+    const messageBody = {
+      job,
+      params: {
+        model_id: model.ref_id,
+        run_id: runId
       }
-    } catch (e) {
-      console.error(e);
-      return Promise.reject();
-    }
+    };
+
+    await this.sqs.send(
+      new SendMessageCommand({
+        MessageBody: JSON.stringify(messageBody),
+        QueueUrl: process.env.JOB_QUEUE_URL,
+        MessageGroupId: "Alfalfa"
+      })
+    );
+
+    return { runId };
   };
 
-  setAlias = async (alias, refId) => {
-    try {
-      const site = await this.sites.findOne({ ref_id: refId });
-      if (site) {
-        return await this.aliases.updateOne(
-          { name: alias },
-          { $set: { name: alias, ref_id: refId } },
-          { upsert: true }
-        );
-      }
-    } catch (e) {
-      console.error(e);
-      return Promise.reject();
-    }
+  setAlias = async (alias, run) => {
```

Comment:
Aliases now use the `ObjectID` of a run instead of it's `ref_id` this is intended to make the look ups more performant as `ref_id` may not be indexed.